app/models/mixins/miq_web_server_runner_mixin.rb

```diff
@@ -33,6 +28,10 @@ def start_rails_server(options)
       Dir.chdir(Vmdb::Application.root)
       server.start
     end
+  rescue SignalException => e
+    raise unless MiqWorker::Runner::INTERRUPT_SIGNALS.include?(e.message)
+  ensure
+    @worker_should_exit = true
```

Comment:
ah, cool, this is the workaround for the kernal trap not working for puma processes, right?

Reply:
Yup, this and the thread join. As it turns out puma returns cleanly when sent a `SIGINT` and re-raises a `SIGTERM` so we catch the signal here, set the instance variable, and join the thread.